include/language-factory.php

```diff
@@ -98,29 +98,31 @@ public static function get_from_terms( array $terms ) {
 		$flag_props = self::get_flag( $data['flag_code'], $data['name'], $data['slug'], $data['locale'] );
 		$data       = array_merge( $data, $flag_props );
 
-		$default = array(
-			'home_url'   => '',
-			'search_url' => '',
-		);
-
+		$additional_data = array();
 		/**
-		 * Filters the home URL and the search URL before the language is created.
+		 * Filters additional data to add to the langage before it will be created.
 		 *
-		 * @since 3.4
+		 * `home_url`, `search_url`, `page_on_front` and `page_for_posts` are only allowed.
 		 *
-		 * @param array  $default {
-		 *     Array of filterable language data.
+		 * @since 3.4
 		 *
-		 *     @type string $home_url   Language home URL.
-		 *     @type string $search_url Language search URL.
-		 * }
+		 * @param array $additional_data.
 		 * @param array $data Language data.
 		 *
-		 * @phpstan-param array{home_url: string, search_url: string} $default
+		 * @phpstan-param array $additional_data
 		 * @phpstan-param non-empty-array<string, mixed> $data
```

Comment:
```suggestion
		 * @phpstan-param array<non-empty-string, mixed> $additional_data
		 * @phpstan-param non-empty-array<non-empty-string, mixed> $data
```

Reply:
Done in 514f8d5759de010583cea8cdafd9342065b8e740